### PR TITLE
!info command

### PR DESCRIPTION
**Pull request by LW001**
So I made a !info command which takes either Username#Discrim, a mention or even a discord ID and gives you info on that user on UV. It even lets you get their UV profile by Discord User ID if they arent on the server.
Even better it shouldn't have syntax errors because I used Visual Studio now
Even better I might add it showing stuff like the latest Upvote and Suggestion
EEEEEven better you get to test it because I can't :^)

### CODE REVIEW

File: Commands/Commands/uservoice.js [hunk 1]
```diff
@@ -241,6 +241,79 @@ commands.comment = {
...
+            let avatar = entities.decode(data.user.avatar_url)
+            let fbUpvoted = entities.decode(data.user.supported_suggestions_count)
+            let fbGiven = entities.decode(data.user.created_suggestions_count)
+            let joinedUV = entities.decode(data.user.createdAt)
+            let dcuser = "*Cannot grab user.*"
+            members.forEach(function(member) {
+              if (userid == member.id) let dcuser = `${member.username}#${member.discriminator}`
+            })
+            msg.channel.sendMessage(`Information for User ${name}:`, false, {
+              color: fc9822,
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: This should be an integer

Comment (Dougley) [reply Commands/Commands/uservoice.js]: Agreed, this isn't a valid color

File: Commands/Commands/uservoice.js [hunk 2]
```diff
@@ -241,6 +241,79 @@ commands.comment = {
...
+    }
+		else msg.channel.sendMessage("Invalid usage. Use a mention, ID or Username#Discriminator combination.").then(errMsg => {
+		setTimeout(bot.deleteMessages([msg,errMsg], msg.channel)),config.timeouts.errorMessageDelete
+		})
+		if (userid != null) {
+			getMail(uvClient,userid).then(email => {
+				uvClient.v1.loginAs(email).then(c => {
+					c.get(`users/${userid}.json`).then(data => {
+						let name = entities.decode(data.user.name)
+            let link = entities.decode(data.user.url)
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: Indent more pls

File: Commands/Commands/uservoice.js [hunk 3]
```diff
@@ -241,6 +241,79 @@ commands.comment = {
...
+            let uvid = entities.decode(data.user.id)
+            let avatar = entities.decode(data.user.avatar_url)
+            let fbUpvoted = entities.decode(data.user.supported_suggestions_count)
+            let fbGiven = entities.decode(data.user.created_suggestions_count)
+            let joinedUV = entities.decode(data.user.createdAt)
+            let dcuser = "*Cannot grab user.*"
+            members.forEach(function(member) {
+              if (userid == member.id) let dcuser = `${member.username}#${member.discriminator}`
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: `let` can't be in a single line if statement

File: Commands/Commands/uservoice.js [hunk 4]
```diff
@@ -241,6 +241,79 @@ commands.comment = {
...
+		else if (suffix.match(/(.{2,32}#\d{4})/)) {
+			let namedisc = suffix.match(/(.{2,32})#(\d{4})/)
+			let members = msg.guild.members
+			let selectedMember = null
+			members.forEach(function(member) {
+				if (member.username == namedisc[1] && member.discriminator == namedisc[2]) userid = member.id
+			})
+    }
+		else msg.channel.sendMessage("Invalid usage. Use a mention, ID or Username#Discriminator combination.").then(errMsg => {
+		setTimeout(bot.deleteMessages([msg,errMsg], msg.channel)),config.timeouts.errorMessageDelete
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: One ) too many, one too short

`setTimeout(bot.deleteMessages([msg,errMsg], msg.channel), config.timeouts.errorMessageDelete)`

**Review by ThaTiemsz — COMMENTED** [review]
­

File: Commands/Commands/uservoice.js (hunk 4 above)
Comment (Dougley) [reply Commands/Commands/uservoice.js]: No, you can't bulk delete a single message

Comment (ThaTiemsz) [reply Commands/Commands/uservoice.js]: Yeah. I was only fixing the syntax, didn't really read the code tbh, it's late ok

File: Commands/Commands/uservoice.js [hunk 5]
```diff
@@ -241,6 +241,81 @@ commands.comment = {
   }
 }
 
+commands.info = {
+  modOnly: false,
+  adminOnly: false,
+  fn: function (bot, msg, suffix, uvClient, cBack) {
+    msg.channel.sendTyping()
+    let mentions = msg.mentions()
```
Comment (Dougley) [inline Commands/Commands/uservoice.js]: `msg.mentions` is not a function, it's an attribute.

**Review by Dougley — CHANGES_REQUESTED** [review]

File: Commands/Commands/uservoice.js [hunk 6]
```diff
@@ -255,46 +255,52 @@ commands.info = {
...
+            let upLink
+            c.get(`suggestions/${latestUpSuggID}`).then(suggestion => {
+              latestUpSuggTitle = suggestion.decode(suggestion.title)
+              upLink = suggestion.decode(suggestion.url)
+            })
+            let latestSuggTitle
+            let latestSuggLink
+            c.get(`users/${userid}/suggestions.json?per_page=1&sort=newest`).then(latestSugg => {
+              latestSuggLink = latestSugg.decode(latestSugg.suggestion.url)
+              latestSuggTitle = latestSugg.decode(latestSugg.suggestion.title)
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: Same here.

File: Commands/Commands/uservoice.js [hunk 7]
```diff
@@ -255,46 +255,52 @@ commands.info = {
...
               }
             })
-            msg.channel.sendMessage(`Information for User ${name}:`, false, {
+            let vForumsArr = entities.decode(data.user.visibleForums)
+            let latestUpSuggID = vForumsArr[vForumsArr.length - 1]
+            let latestUpSuggTitle = `*No suggestion found*`
+            let upLink
+            c.get(`suggestions/${latestUpSuggID}`).then(suggestion => {
+              latestUpSuggTitle = suggestion.decode(suggestion.title)
+              upLink = suggestion.decode(suggestion.url)
```
Comment (ThaTiemsz) [inline Commands/Commands/uservoice.js]: This most likely won't work.

**Review by ThaTiemsz — COMMENTED** [review]
See comments I guess

**Review by Dougley — APPROVED** [review]

**Comment by Dougley** [comment]
Squashing this into develop

**Pull request merged**